extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/ZipkinReporterProvider.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.jaeger.runtime;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+import javax.interceptor.Interceptor;
+
+import io.quarkus.arc.AlternativePriority;
+import io.quarkus.arc.properties.UnlessBuildProperty;
+
+@Singleton
+public class ZipkinReporterProvider {
+    @Produces
+    @Singleton
+    @UnlessBuildProperty(name = "quarkus.jaeger.zipkin.compatibility-mode", stringValue = "false", enableIfMissing = false)
```

Comment:
Is this needed, or is the processor guard to make it unremovable enough?

Reply:
its enough the processor guard, I removed it.
After all the changes I tested in a new application (jvm and native mode) the following scenarios:

- no dependency and zipkin mode disabled -> the span was delivered in jaeger
- no dependency and zipkin mode enabled -> the span was delivered in jaeger
- with dependency and zipkin mode enabled -> the span was delivered in zipkin
- with dependency and zipkin mode disabled -> the span was delivered in jaeger